src/engine/shared/console.cpp

```diff
@@ -667,9 +668,52 @@ static void StrVariableCommand(IConsole::IResult *pResult, void *pUserData)
 		char aBuf[1024];
 		str_format(aBuf, sizeof(aBuf), "Value: %s", pData->m_pStr);
 		pData->m_pConsole->Print(IConsole::OUTPUT_LEVEL_STANDARD, "Console", aBuf);
+		str_copy(pResult->m_aValue, pData->m_pStr, sizeof(pResult->m_aValue));
 	}
 }
 
+void CConsole::Con_EvalIf(IResult *pResult, void *pUserData)
+{
+	CCommand *pCommand = ((CConsole*)pUserData)->FindCommand(pResult->GetString(0), ((CConsole*)pUserData)->m_FlagMask);
+	char aBuf[128];
+	if(!pCommand)
+	{
+		str_format(aBuf, sizeof(aBuf), "No such command: '%s'.", pResult->GetString(0));
+		((CConsole*)pUserData)->Print(OUTPUT_LEVEL_STANDARD, "Console", aBuf);
+		return;
+	}
+	CResult Result;
+	pCommand->m_pfnCallback(&Result, pCommand->m_pUserData);
+	bool Condition = false;
+	if(pCommand->m_pfnCallback == IntVariableCommand)
+		Condition = Result.m_Value == atoi(pResult->GetString(2));
+	else
+		Condition = !str_comp_nocase(Result.m_aValue, pResult->GetString(2));
+	if(!str_comp_num(pResult->GetString(1), "!=", 3))
+		Condition = !Condition;
+	else if(str_comp_num(pResult->GetString(1), "==", 3) && pCommand->m_pfnCallback == StrVariableCommand)
+		((CConsole*)pUserData)->Print(OUTPUT_LEVEL_STANDARD, "Console", "Error: invalid comperator for type string");
+	else if(!str_comp_num(pResult->GetString(1), ">", 2))
+		Condition = Result.m_Value > atoi(pResult->GetString(2));
+	else if(!str_comp_num(pResult->GetString(1), "<", 2))
+		Condition = Result.m_Value < atoi(pResult->GetString(2));
+	else if(!str_comp_num(pResult->GetString(1), "<=", 3))
+		Condition = Result.m_Value <= atoi(pResult->GetString(2));
+	else if(!str_comp_num(pResult->GetString(1), ">=", 3))
+		Condition = Result.m_Value >= atoi(pResult->GetString(2));
+	else if(str_comp_num(pResult->GetString(1), "==", 3))
+		((CConsole*)pUserData)->Print(OUTPUT_LEVEL_STANDARD, "Console", "Error: invalid comperator for type integer");
+
+	if(pResult->NumArguments() > 4 && str_comp_num(pResult->GetString(4), "else", 5))
+		((CConsole*)pUserData)->Print(OUTPUT_LEVEL_STANDARD, "Console", "Error: expected else");
+
+	CConsole* pConsole = static_cast<CConsole *>(pUserData);
```

Comment:
Could move that line on top, so you don't need the casts(`((CConsole*)pUserData)`) all the time